[Build] Compilation error with clang-9 and c++17 due to incompatible aggregate initialization pattern in ewasm backend code

## Description

https://github.com/ethereum/solidity/blob/6c31a5f3ddfb40d1f5f70823569d2f0c6792d13c/libyul/backends/wasm/EWasmCodeTransform.cpp#L58

results in the following compilation error

```
libyul/backends/wasm/EWasmCodeTransform.cpp:58:21: error: temporary of type 'boost::static_visitor<std::string>' (aka 'static_visitor<basic_string<char,
 char_traits<char>, allocator<char> > >') has protected destructor
        return EWasmToText{}.run(
                           ^
/usr/local/include/boost/variant/static_visitor.hpp:53:5: note: declared protected here
    ~static_visitor() = default;
    ^
```

because, in c++17, classes with explicitly defaulted constructors are considered aggregates

> no user-provided, inherited, or explicit constructors (explicitly defaulted or deleted constructors are allowed)

See https://en.cppreference.com/w/cpp/language/aggregate_initialization#Explanation

This means that the aggregate base class (`boost::static_visitor` in this case) is initialized via aggregate initialization (emphasis mine)

> If the number of initializer clauses is less than the number of members and bases (since C++17) or initializer list is completely empty, the remaining members and bases (since C++17) are initialized by their default member initializers, if provided in the class definition, and **otherwise (since C++14) by empty lists**, in accordance with the usual list-initialization rules (which performs value-initialization for non-class types and non-aggregate classes with default constructors, and **aggregate initialization for aggregates**)

As noted by this resourceful Stackoverflow contributor (See https://stackoverflow.com/a/56745475), this means that the base class constructor is bypassed by the aggregate initialization. However, when the base object is to be destroyed, the compiler encounters a protected destructor (in this case for `boost::static_visitor`) and hence the compilation error.

## Environment

- Compiler version: latest develop
- Operating system: Compiler used is clang-9 in c++17 mode

## Steps to Reproduce

Compile solidity using clang-9. clang-8 might also demonstrate the error going by OP here https://stackoverflow.com/a/56745475

--- a/libyul/backends/wasm/EWasmCodeTransform.cpp
+++ b/libyul/backends/wasm/EWasmCodeTransform.cpp
@@ -55,7 +55,7 @@ string EWasmCodeTransform::run(Dialect const& _dialect, yul::Block const& _ast)
 	std::vector<wasm::FunctionImport> imports;
 	for (auto& imp: transform.m_functionsToImport)
 		imports.emplace_back(std::move(imp.second));
-	return EWasmToText{}.run(
+	return EWasmToText().run(
 		transform.m_globalVariables,
 		imports,
 		functions

